test/unit/mem/view/src/Accessor.cpp

```diff
@@ -0,0 +1,337 @@
+/* Copyright 2021 Bernhard Manfred Gruber
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+#include <alpaka/mem/view/Accessor.hpp>
+#include <alpaka/mem/view/ViewPlainPtr.hpp>
+#include <alpaka/mem/view/ViewStdArray.hpp>
+#include <alpaka/mem/view/ViewStdVector.hpp>
+#include <alpaka/mem/view/ViewSubView.hpp>
+
+#include <catch2/catch.hpp>
+
+TEST_CASE("isView", "[accessor]")
+{
+    using alpaka::traits::internal::isView;
+
+    using Dim = alpaka::DimInt<1>;
+    using Size = std::size_t;
+    using Acc = alpaka::ExampleDefaultAcc<Dim, Size>;
+    using Dev = alpaka::Dev<Acc>;
+
+    // buffer
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto buffer = alpaka::allocBuf<int, Size>(devAcc, Size{1});
+    STATIC_REQUIRE(isView<decltype(buffer)>);
+
+    // views
+    STATIC_REQUIRE(isView<alpaka::ViewPlainPtr<Dev, int, Dim, Size>>);
+    STATIC_REQUIRE(isView<std::array<int, 42>>);
+    STATIC_REQUIRE(isView<std::vector<int>>);
+    STATIC_REQUIRE(isView<alpaka::ViewSubView<Dev, int, Dim, Size>>);
+
+    // accessor
+    auto accessor = alpaka::access(buffer);
+    STATIC_REQUIRE(!isView<decltype(accessor)>);
+}
+
+namespace
+{
+    constexpr auto N = 1024;
+
+    struct WriteKernelTemplate
+    {
+        template<typename TAcc, typename TAccessor>
+        ALPAKA_FN_ACC void operator()(TAcc const&, TAccessor data) const
+        {
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::DimInt<1>, alpaka::Idx<TAcc>>{alpaka::Idx<TAcc>{3}}] = 3.0f;
+        }
+    };
+
+    struct WriteKernelExplicit
+    {
+        template<typename TAcc, typename TMemoryHandle, typename TIdx>
+        ALPAKA_FN_ACC void operator()(
+            TAcc const&,
+            alpaka::Accessor<TMemoryHandle, float, TIdx, 1, alpaka::WriteAccess> const data) const
+        {
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::DimInt<1>, TIdx>{TIdx{3}}] = 3.0f;
+        }
+    };
+
+    struct ReadKernelTemplate
+    {
+        template<typename TAcc, typename TAccessor>
+        ALPAKA_FN_ACC void operator()(TAcc const&, TAccessor data) const
+        {
+            const float v1 = data[1];
```

Comment:
Remember `east const`!

Reply:
I am still waiting for an automated tool to do that. Unfortunately, the progress in clang-format is very slow.